rust_src/src/buffers.rs

```diff
@@ -340,6 +348,14 @@ impl LispBufferRef {
             None
         }
     }
+
+    pub fn set_value(&mut self, offset: usize, value: LispObject) {
```

Comment:
This function doesn't check that the offset is inside the buffer, so it should be `unsafe`.